Replicator/ChangesFeed.cc

```diff
@@ -137,13 +132,6 @@ namespace litecore::repl {
             uint32_t nChanges = nextObservation.numChanges;
             if ( nChanges == 0 ) break;
 
-            if ( !nextObservation.external && !_echoLocalChanges ) {
-                logDebug("Observed %u of my own db changes #%" PRIu64 " ... #%" PRIu64 " (ignoring)", nChanges,
-                         static_cast<uint64_t>(c4changes[0].sequence),
-                         static_cast<uint64_t>(c4changes[nChanges - 1].sequence));
-                _maxSequence = c4changes[nChanges - 1].sequence;
-                continue;  // ignore changes I made myself
```

Comment:
intentional changes?

Reply:
Oh! That's a good catch. I thought for a while I had to get rid of that optimization because it wouldn't work with the database pool; but actually it does. I'll put that code back in.

Reply:
Actually there was a reason behind that, as I rediscovered when I put it back. That code block is an optimization to ignore "echoes": changes made by a corresponding puller. However it had been nonfunctional for years (the 'external' flag would never be false because it's observing a C4Database that's never used for writes.) With the DatabasePool changes, that code now works, but (a) it exposes a bug somewhere in the pusher that doesn't update the checkpoint, thus the test failure; and (b) it's not going to work correctly in Edge Server where everything is sharing that writeable db.

tl;dr: I took it out again and it fixed the test. I left an explanatory comment in its place.